### PR TITLE
Feature/implement the forgot UI

### LINKED ISSUE #5: Implement the forgot UI
Image

### PR DESCRIPTION
**Pull request by KMTus**
## Related issues
Resolved #5 

## Summary
- Create template forgot password UI light-dark mode 
- Added new translation keys and new color codes
- Added color at interface and constranst
- Added library Expo LinearGradient  
### step 
1. Open app/_layout.tsx 
2. Commet ``` <TamaguiProvider />```
3. Import tempalte ```import ForgotTemplate from '~/components/templates/PorgotPasswordTemplate'``` 
4. Call ```<ForgotTemplate/>```

## Screenshots

image
image

### CODE REVIEW

File: components/atoms/NegativeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
+import { useColorScheme } from 'react-native'
+import { Button, type ButtonProps } from 'tamagui'
+
+import getColors from '~/constants/Colors'
+import { RADIUS_BUTTON } from '~/constants/Constants'
+
+type Props = {
+  title: string
+  icon?: JSX.Element
```
Comment (NguyenVanViet2004) [inline components/atoms/NegativeButton.tsx:10]: The icon property is already available in [ButtonProps](https://tamagui.dev/ui/button#api-reference), there is no need to add an icon property here. Delete it

```suggestion

  type Props = {
  title: string
  } & ButtonProps

```

File: components/atoms/TextTitle.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  text: string
+} & TextProps
+
+export const TextTitle = (props: Props): React.ReactElement => {
+  const colors = getColors(useColorScheme())
+  return (
+    <Text
+      {...props}
+      fontSize={14}
+      fontFamily={'JetBrainsMonoBold'}
```
Comment (NguyenVanViet2004) [inline components/atoms/TextTitle.tsx:17]: use hook [`useAppFont()`](https://github.com/NguyenVanViet2004/FE-GlowUp/blob/main/components/templates/HomeTemplate.tsx) for set fontFamily

File: components/molecules/common/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        color={colors.slateGray}
+        textAlign="center">
+        {title}
+      </Text>
+
+      <Text
+        onPress={onPressAuthScreen}
+        fontSize={14}
+        fontFamily={'JetBrainsMonoBold'}
+        color={colors.primaryTeal}
```
Comment (NguyenVanViet2004) [inline components/molecules/common/Footer.tsx:31]: `Footer` for what? Rename

```suggestion
        color={colors.primaryTeal} />
```

File: components/templates/PorgotPasswordTemplate.tsx
Comment (NguyenVanViet2004) [inline components/templates/PorgotPasswordTemplate.tsx:1]: `Forgot `not `Porgot`

File: components/templates/PorgotPasswordTemplate.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import { StyleSheet, useColorScheme } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { TamaguiProvider, YStack } from 'tamagui'
+
+import ContentTitle from '~/components/atoms/ContentTitle'
+import getColors from '~/constants/Colors'
+import useTranslation from '~/hooks/useTranslation'
+import config from '~/tamagui.config'
+
+import InputWithIcons from '../atoms/InputWithIcons'
```
Comment (NguyenVanViet2004) [inline components/templates/PorgotPasswordTemplate.tsx:13]: Relative path

**Review by NguyenVanViet2004 — CHANGES_REQUESTED** [review]

File: components/atoms/NegativeButton.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+} & ButtonProps
+
+export const NegativeButton = (props: Props): React.ReactElement => {
+  const colors = getColors(useColorScheme())
+  return (
+    <Button
```
Comment (thanhdo29) [inline components/atoms/NegativeButton.tsx:16]: Missing pressStyle property to change color when pressing Button

File: components/molecules/InputForm.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+type Props = {
+  visibleRecoveryPassword?: boolean
+  visibleInputWithIcons?: boolean
+  visibleForgotPassword?: boolean
+  onChangeNameText?: (text: string) => void
+  onChangeEmailText?: (text: string) => void
+  onChangePhoneText?: (text: string) => void
+  onChangePasswordText?: (text: string) => void
+} & InputProps
```
Comment (thanhdo29) [inline components/molecules/InputForm.tsx:19]: I think there is an excess of InputProps here

**Review by thanhdo29 — APPROVED** [review]

**Review by thanhdo29 — DISMISSED** [review]

File: components/templates/PorgotPasswordTemplate.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  const IconVisibleForgot = isPhoneEmail
+    ? <Mail size={16} color={colors.primaryTeal} />
+    : <Phone size={16} color={colors.primaryTeal} />
+
+  return (
+    <TamaguiProvider config={config} defaultTheme={colorScheme as any}>
+      <SafeAreaView style={styles.flex}>
+        {colorScheme === 'dark'
+          ? (
+            <LinearGradient
```
[the remark this answers is not in the document]
Comment (KMTus) [reply components/templates/PorgotPasswordTemplate.tsx]: Surely, I will do that

**Comment by NguyenVanViet2004** [comment]
Fix it

Simulator Screenshot - iPhone 15 Pro - 2024-09-26 at 22 04 06

**Comment by truongtdph40209** [comment]
image

**Review by truongtdph40209 — APPROVED** [review]

**Review by thanhdo29 — APPROVED** [review]

**Review by truongtdph40209 — APPROVED** [review]

**Review by thanhdo29 — APPROVED** [review]

**Comment by NguyenVanViet2004** [comment]
@KMTus Fix conflicts

**Review by truongtdph40209 — APPROVED** [review]

**Review by thanhdo29 — APPROVED** [review]

**Review by NguyenVanViet2004 — APPROVED** [review]

**Pull request merged**